Maple2Storage/Types/Metadata/MasteryMetadata.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Serialization;
+
+namespace Maple2Storage.Types.Metadata
+{
+    [XmlType]
+    public class MasteryMetadata
+    {
+        [XmlElement(Order = 1)]
+        public int Type;
+        [XmlElement(Order = 2)]
+        public List<MasteryGradeMetadata> Grades;
+
+        // Required for deserialization
+        public MasteryMetadata()
+        {
+            Grades = new List<MasteryGradeMetadata>();
+        }
+
+        public MasteryMetadata(int type, List<MasteryGradeMetadata> grades)
+        {
+            Type = type;
+            Grades = grades;
+        }
+
+        public override string ToString() =>
+            $"MasteryMetadata(Type:{Type},Grades:{string.Join(",", Grades)})";
+
+        protected bool Equals(MasteryMetadata other)
+        {
+            return Type == other.Type
+                   && Grades.SequenceEqual(other.Grades);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null)
+                return false;
+            if (ReferenceEquals(this, obj))
+                return true;
+            if (obj.GetType() != GetType())
+                return false;
+            return Equals((MasteryMetadata) obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(Type);
+        }
+
+        public static bool operator ==(MasteryMetadata left, MasteryMetadata right)
+        {
+            return Equals(left, right);
+        }
+
+        public static bool operator !=(MasteryMetadata left, MasteryMetadata right)
+        {
+            return !Equals(left, right);
+        }
+    }
+
+    [XmlType]
+    public class MasteryGradeMetadata
```

Comment:
Usually only the top level object uses the term `Metadata` to keep from any confusion, so prefer to rename to `MasteryGrade` here.